pactman/verifier/pytest_plugin.py

```diff
@@ -1,7 +1,111 @@
-from . import broker_pact
+import glob
+import logging
+import pytest
+import os
+from _pytest.outcomes import Failed
+
+from .broker_pact import BrokerPact, BrokerPacts
+from .result import log, PytestResult
+
+
+def pytest_addoption(parser):
+    parser.addoption("--pact-files", default=None,
+                     help="pact JSON files to verify (wildcards allowed)")
+    parser.addoption("--pact-broker-url", default='',
+                     help="pact broker URL")
+    parser.addoption("--pact-provider-name", default=None,
+                     help="provider pact name")
+    parser.addoption("--pact-publish-results", action="store_true", default=False,
+                     help="report pact results to pact broker")
+    parser.addoption("--pact-provider-version", default=None,
+                     help="provider version to use when reporting pact results to pact broker")
+
+
+def get_broker_url(config):
+    if config.getoption('pact_broker_url'):
+        return config.getoption('pact_broker_url')
+    if os.environ.get('PACT_BROKER_URL'):
+        return os.environ.get('PACT_BROKER_URL')
+    return None
 
 
 # add the pact broker URL to the pytest output if running verbose
 def pytest_report_header(config):
     if config.getoption('verbose') > 0:
-        return [f'Loading pacts from {broker_pact.PACT_BROKER_URL}']
+        location = get_broker_url(config) or config.getoption('pact_files')
+        return [f'Loading pacts from {location}']
+
+
+def pytest_configure(config):
+    logging.getLogger().handlers = []
+    logging.basicConfig(format='%(message)s')
+    verbosity = config.getoption('verbose')
+    if verbosity > 0:
+        log.setLevel(logging.DEBUG)
+
+
+class PytestPactVerifier:
+    def __init__(self, publish_results, provider_version, interaction, consumer):
+        self.publish_results = publish_results
+        self.provider_version = provider_version
+        self.interaction = interaction
+        self.consumer = consumer
+
+    def verify(self, provider_url, provider_setup):
+        try:
+            self.interaction.verify_with_callable_setup(provider_url, provider_setup)
+        except (Failed, AssertionError) as e:
+            raise Failed(str(e)) from None
+
+    def finish(self):
+        if self.consumer and self.publish_results and self.provider_version:
+            self.consumer.publish_result(self.provider_version)
+
+
+def flatten_pacts(pacts):
+    for consumer in pacts:
+        last = consumer.interactions[-1]
+        for interaction in consumer.interactions:
+            if interaction is last:
+                yield (interaction, consumer)
+            else:
+                yield (interaction, None)
+
+
+def get_pact_files(file_location):
+    if not file_location:
+        return []
+    for filename in glob.glob(file_location):
+        yield BrokerPact.load_file(filename, result_factory=PytestResult)
+
+
+def test_id(identifier):
+    interaction, consumer = identifier
```

Comment:
_ instead of consumer since it is ignored ? 

OR 

`return str(identifier[0])`